contracts/BandOracle.cdc

```diff
@@ -94,48 +95,60 @@ pub contract BandOracle {
     ///
-    pub resource interface OracleAdmin {
-        pub fun getUpdaterCapabilityPathFromAddress (relayer: Address): PrivatePath
-        pub fun removeSymbol (symbol: String)
-        pub fun createNewFeeCollector (): @BandOracle.FeeCollector
+    access(all) resource interface OracleAdmin {
+        access(all) fun setRelayerCapabilityID (relayer: Address, capabilityID: UInt64)
+        access(all) fun revokeRelayer (relayer: Address)
+        access(all) fun getUpdaterCapabilityIDFromAddress (relayer: Address): UInt64?
+        access(all) fun removeSymbol (symbol: String)
+        access(all) fun createNewFeeCollector (): @BandOracle.FeeCollector
     }
 
     /// Relayer operations.
     ///
-    pub resource interface DataUpdater {
-        pub fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
+    access(all) resource interface DataUpdater {
+        access(all) fun updateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
                             requestID: UInt64, relayerID: UInt64)
-        pub fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
+        access(all) fun forceUpdateData (symbolsRates: {String: UInt64}, resolveTime: UInt64, 
                             requestID: UInt64, relayerID: UInt64)
     }
 
     /// The `BandOracleAdmin` will be created on the contract deployment, and will allow 
     /// the own admin to manage the oracle and the relayers to update prices on it.
     ///
-    pub resource BandOracleAdmin: OracleAdmin, DataUpdater {
+    access(all) resource BandOracleAdmin: OracleAdmin, DataUpdater {
 
-        /// Auxiliary method to ensure that the formation of the capability path that 
-        /// identifies relayers is done in a uniform way.
+        /// Stores in contract the data updater capability ID along with the address
+        /// of the relayer who got the capability
         ///
         /// @param relayer: The entitled relayer account address
+        /// @param capabilityID: The ID of the data updater capability
         ///
-        pub fun getUpdaterCapabilityPathFromAddress (relayer: Address): PrivatePath {
-            // Create the string that will form the private path concatenating the base
-            // path and the relayer identifying address
-            let privatePathString = 
-                BandOracle.getUpdaterCapabilityNameFromAddress(relayer: relayer)
-            // Attempt to create the private path using the identifier
-            let dataUpdaterPrivatePath = 
-                PrivatePath(identifier: privatePathString) 
-                ?? panic("Error while creating data updater capability private path")
-            return dataUpdaterPrivatePath
+        access(all) fun setRelayerCapabilityID (relayer: Address, capabilityID: UInt64) {
+            BandOracle.relayersCapabilityID[relayer] = capabilityID
+        }
+
+        /// Deletes a relayer's data updater capability from the contract storage
```

Comment:
All this really does is remove the id from the mapping. Even after executing, the capability could still be active. IMO we should either add a `Capability<auth(StorageCapabilities) &Account>` to the `BandOracleAdmin` and control capability issuance/deletion directly from the resource or update the code docs here to mention that deletion would need to be done separately. My preference would be the former, but if time constraints demand a quick solution, here's the suggestion for the latter.

```suggestion
        /// Deletes a relayer's CapabilityID from 
```

The former interface would looks something like:

```cadence
/// Issues a DataUpdater Capability and associates the Capability ID with the provided relayer address
access(all) fun issueRelayerCapability(relayer: Address): Capability<&{DataUpdater}>
/// Revokes the relayer's issued capability, determining the ID from the contract's `relayersCapabilityID` mapping
access(all) fun revokeRelayer (relayer: Address)
```

and then ofc the admin would need the account capability on init

Reply:
yeah that is done in a transaction atm, probably revokeRelayer is a bad name for the method

`transactions/OracleAdmin/revoke_relayer.cdc`
```
if let capabilityID = oracleAdminRef.getUpdaterCapabilityIDFromAddress(relayer: relayer){
            let controller = acct.capabilities.storage.getController(byCapabilityID: capabilityID) ??
                panic("Can't get relayer's data updater capability controller")
            controller.delete()
            oracleAdminRef.revokeRelayer (relayer: relayer)
```

Reply:
I added your comment and updated the function name. I agree that having a method inside Admin to handle this will be the way to go. If I'm understanding it correctly, this way will need the admin resource to store a capability to the account storage right? Given the time frame probably just keep it this way and make sure that band understand how it works.